sdk/aqueduct/artifacts/system_metric.py

```diff
@@ -0,0 +1,115 @@
+import uuid
+from typing import List, Union
+
+from aqueduct.artifacts import bool_artifact, numeric_artifact
+from aqueduct.artifacts import preview as artifact_utils
+from aqueduct.artifacts._create import create_metric_or_check_artifact
+from aqueduct.constants.enums import ArtifactType, ExecutionMode
+from aqueduct.constants.metrics import SYSTEM_METRICS_INFO
+from aqueduct.models.artifact import ArtifactMetadata
+from aqueduct.models.dag import DAG
+from aqueduct.models.operators import Operator, OperatorSpec, SystemMetricSpec
+from aqueduct.utils.naming import default_artifact_name_from_op_name
+from aqueduct.utils.utils import generate_uuid
+
+
+class SystemMetricMixin:
+    """A mixin class for the system_metric function. This is used by GenericArtifacts and TableArtifacts."""
+
+    def list_system_metrics(self) -> List[str]:
+        """Returns a list of all system metrics available on the table artifact.
+        These system metrics can be set via the invoking the system_metric() method the table.
+
+        Returns:
+            A list of available system metrics on a table
+        """
+        return list(SYSTEM_METRICS_INFO.keys())
```

Comment:
I wonder if we should return the descriptions along with each system metric name? Essentially, keeping it a dictionary. That sees more useful to the user than just a list of system metric names.